.github/workflows/release.yml

```diff
@@ -0,0 +1,130 @@
+        with:
+          gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
+          gitlab_user: ${{ secrets.GITLAB_USER }}
+          gitlab_pass: ${{ secrets.GITLAB_PASS }}
+
+      - name: Caching
+        uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191
+        with:
+          cache-on-failure: true
+          cache-all-crates: true
+
+      - name: Install additional cargo tooling
+        uses: taiki-e/cache-cargo-install-action@3d5e3efe44b020826abe522d18cb4457042280ef
+        with:
+          tool: cargo-auditable
+
+      - name: Build release
+        shell: bash
+        run: cargo auditable build --release
+
+      - name: Rename binary
+        shell: bash
+        run: "mv target/release/${{ github.event.repository.name }} target/release/${{ matrix.asset_name }}"
+
+      - name: Attest
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-path: '${{ github.workspace }}/target/release/${{ matrix.asset_name }}'
+
+      - name: Upload binary
+        uses: actions/upload-artifact@v4
+        with:
+          name: release-${{ matrix.asset_name }}
+          path: '${{ github.workspace }}/target/release/${{ matrix.asset_name }}'
+
+  sbom:
+    permissions:
+      id-token: write
+      contents: read
+      attestations: write
+    runs-on: ubuntu-latest
+    container: docker-oss.nexus.famedly.de/rust-container:nightly
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: famedly/backend-build-workflows/.github/actions/rust-prepare@main
+        with:
+          gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
+          gitlab_user: ${{ secrets.GITLAB_USER }}
+          gitlab_pass: ${{ secrets.GITLAB_PASS }}
+
+      - name: Caching
+        uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191
+        with:
+          cache-on-failure: true
+          cache-all-crates: true
+
+      - name: Install additional cargo tooling
+        uses: taiki-e/cache-cargo-install-action@3d5e3efe44b020826abe522d18cb4457042280ef
+        with:
+          tool: cargo-sbom
+
+      - name: Generate SBOM
+        shell: bash
+        run: cargo sbom > sbom.spdx.json
+
+      - name: Attest
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-path: '${{ github.workspace }}/sbom.spdx.json'
+
+      - name: Upload SBOM
+        uses: actions/upload-artifact@v4
+        with:
+          name: release-sbom-spdx
+          path: '${{ github.workspace }}/sbom.spdx.json'
+
+  release:
+    runs-on: ubuntu-latest
+    needs: [build, sbom]
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: release-*
+          path: artifacts
+          merge-multiple: true
+
+      - name: Create release
+        uses: softprops/action-gh-release@79721680dfc87fb0f44dfe65df68961056d55c38
+        with:
+          files: artifacts/*
```

Comment:
Could be nice to inspect the triggering tag and mark the release as a pre-release depending on whether the tag includes `rc.X`

Reply:
done